google/cloud/stream_range.h

```diff
@@ -186,22 +196,46 @@ class StreamRange {
   }
 
   template <typename U>
-  friend StreamRange<U> MakeStreamRange(StreamReader<U>);
+  friend StreamRange<U> internal::MakeStreamRange(internal::StreamReader<U>);
 
   /**
    * Constructs a `StreamRange<T>` that will use the given @p reader.
    *
+   * The `T` objects are read from the caller-provided `internal::StreamReader`
+   * functor, which is invoked repeatedly as the range is iterated. The
+   * `internal::StreamReader` can return an OK `Status` to indicate a successful
+   * end of stream, or a non-OK `Status` to indicate an error, or a `T`. The
+   * `internal::StreamReader` will not be invoked again after it returns a
+   * `Status`.
+   *
+   * @par Example: Printing integers from 1-10.
+   *
+   * @code
+   * int counter = 0;
+   * auto reader = [&counter]() -> internal::StreamReader<int>::result_type {
+   *   if (counter++ < 10) return counter;
```

Comment:
Nit for moved example: `if (++counter <= 10) return counter;` is a little nicer.

Reply:
Changed here and above.